src/App.tsx

```diff
@@ -1,39 +1,64 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
 
+import { Clock } from './components/Clock';
+
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  clockName: string;
+  hasClock: boolean;
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerId: number | null = null;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      const clockName = getRandomName();
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+      this.setState({ clockName });
+    }, 3300);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.addEventListener('contextmenu', this.handleContextMenu);
+    document.addEventListener('click', this.handleClick);
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentWillUnmount() {
+    if (this.timerId) {
+      window.clearInterval(this.timerId);
+    }
+  }
```

Comment:
Just like good practice. If you added a custom event - clear it. Always clean up after yourself so that there are no unnecessary side effects

```suggestion
  componentWillUnmount() {
    if (this.timerId) {
      window.clearInterval(this.timerId);
    }
    document.removeEventListener('contextmenu', this.handleContextMenu);
    document.removeEventListener('click', this.handleClick);
  }
```